### DISCUSSION TITLE
[Coil3] Regarding the issue of setting a global OkHttpClient

### DISCUSSION THREAD

**Discussion by azhu003** [coil-kt/coil#2608] [Q&A] [answered]
Hi, I'm trying to use the new API of coil3-compose. In version 2.7.0, I could set Coil's global OkHttp instance by using ImageLoader.Builder.okHttpClient(client), which was very helpful for uniformly handling some requests. However, in version 3.0.0-rc02, I couldn't find a method to set it globally, and I have to modify the model parameter of each AsyncImage. Is there a better way in Coil? I don't want to modify too much of the original code. Thanks
<img width="773" alt="image" src="https://github.com/user-attachments/assets/5589a688-367b-4992-bdd2-83d696d3eec6">

**Comment by colinrtwhite** [comment, marked as answer]
Here's the latest way to set a global OkHttpClient: https://coil-kt.github.io/coil/recipes/#using-a-custom-okhttpclient

**Reply by azhu003** [reply]
Thank you very much. I found it.

**Comment by R9DCHDC** [comment]
The link in the above answer is invalid now. Use this one https://coil-kt.github.io/coil/network/#using-a-custom-okhttpclient
